### PR TITLE
feat: implement asynchronous reports triggering

### PR DESCRIPTION
**Pull request by pmalek**
This PR aims to address #38 which allows users to trigger reports asynchronously, and to provide custom signal names at the call site.

It also adds some documentation on how to use this framework.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,63 @@ Toolkit for telemetry data for [Kong's][kong] Kubernetes products, such as the
...
+  // Start the manager
+  err := m.Start()
+  // Handle errors ...
+
+
+  // Trigger asynchronous report as needed.
+  err := m.TriggerExecute(context.Background(), "custom-event-happened");
+  // Handle errors ...
+```
+
```
Comment (shaneutt) [inline README.md:64]: I think the inclusion of an example go program is a nice touch :+1:

File: pkg/telemetry/manager.go [hunk 2]
```diff
@@ -37,6 +39,9 @@ const (
 )
 
 type manager struct {
+	// TODO
```
Comment (shaneutt) [inline pkg/telemetry/manager.go:42]: What are we doing about this TODO? What is it here for?

File: pkg/telemetry/manager.go [hunk 3]
```diff
@@ -77,18 +83,28 @@ type Manager interface {
 	AddConsumer(c Consumer) error
 	// AddWorkflow adds a workflow with providers which will provide telemetry data.
 	AddWorkflow(Workflow)
-	// Execute executes all workflows and returns an aggregated report from those
+	// TODO
```
Comment (shaneutt) [inline pkg/telemetry/manager.go:86]: Another TODO we should resolve or have an issue linked for.

Comment (pmalek) [reply pkg/telemetry/manager.go]: Leftovers, removed that.

File: pkg/telemetry/manager.go (hunk 2 above)
Comment (pmalek) [reply pkg/telemetry/manager.go]: Leftovers from marking it for myself, removed.

**Review by mlavacca — APPROVED** [review]

**Pull request merged**